### PR TITLE
Feature/publishing catalogi admin

### PR DESCRIPTION
**Pull request by stevenbal**
for https://github.com/open-zaak/open-zaak/issues/194

### CODE REVIEW

File: src/openzaak/templates/admin/submit_line.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{% extends "admin/submit_line.html" %}
+{% load i18n admin_urls %}
+
+{% block submit-row %}
+    {% if original.concept == True %}
+        <input type="submit" value="{% trans 'Publish' %}" name="_publish"/>
+
+        {% comment %} <a href="{% url 'admin:zaaktype-publish' original.id %}">{% trans 'Publish' %}</a> {% endcomment %}
```
Comment (joeribekker) [inline src/openzaak/templates/admin/submit_line.html]: This should not be needed :)

File: src/openzaak/components/catalogi/models/zaaktype.py [hunk 2]
```diff
@@ -358,7 +358,10 @@ class Meta:
         verbose_name_plural = _("Zaaktypen")
 
     def __str__(self):
-        return "{} - {}".format(self.catalogus, self.zaaktype_identificatie)
+        representation = "{} - {}".format(self.catalogus, self.zaaktype_identificatie)
+        if self.concept:
+            representation = "{} (CONCEPT)".format(representation)
+        return representation
```
Comment (joeribekker) [inline src/openzaak/components/catalogi/models/zaaktype.py]: We also need this for Besluittype and InformatieObjecttype

File: src/openzaak/components/catalogi/admin/mixins.py [hunk 3]
```diff
@@ -16,3 +18,21 @@ class ConceptAdminMixin(object):
     def get_fieldsets(self, request, obj=None):
         fieldsets = super().get_fieldsets(request, obj)
         return tuple(fieldsets) + ((_("Concept"), {"fields": ("concept",)}),)
+
+
+class PublishAdminMixin:
+    def response_post_save_change(self, request, obj):
```
Comment (joeribekker) [inline src/openzaak/components/catalogi/admin/mixins.py]: I see that you copied this code for Zaaktype with validation. That's not very DRY.

Why not add a new function like:

```python
def _publish_validation(...):
    pass  # default

def response_post_save_change(self, request, obj):
    if "_publish" in request.POST:
        # ...
        if not self._publish_validation():
            return
        # ...
```

And override `_publish_validation` with the code you added for Zaaktype?

File: src/openzaak/components/catalogi/admin/zaaktypen.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
-from django.contrib import admin
+from django.contrib import admin, messages
```
Comment (sergei-maertens) [inline src/openzaak/components/catalogi/admin/zaaktypen.py]: please clean up the unused imports

**Pull request merged**